From 10d35b5a2eac8aec0eaf640eab24e21534044cfd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Mon, 11 Aug 2014 12:09:16 +0200
Subject: [PATCH] sudo: work with correct D-Bus iterator

---
 src/providers/data_provider_be.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/providers/data_provider_be.c b/src/providers/data_provider_be.c
index 37d1e28c4ca9ae1733496d9bf2b4a955c1896b3a..180f7225457ad3a82b4c100df15032c90b8a8529 100644
--- a/src/providers/data_provider_be.c
+++ b/src/providers/data_provider_be.c
@@ -1578,14 +1578,15 @@ static int be_sudo_handler(struct sbus_request *dbus_req, void *user_data)
 
         /* read the rules */
         for (i = 0; i < rules_num; i++) {
-            if (dbus_message_iter_get_arg_type(&iter) != DBUS_TYPE_STRING) {
+            if (dbus_message_iter_get_arg_type(&array_iter)
+                    != DBUS_TYPE_STRING) {
                 DEBUG(SSSDBG_CRIT_FAILURE, "Failed, to parse the message!\n");
                 ret = EIO;
                 err_msg = "Invalid D-Bus message format";
                 goto fail;
             }
 
-            dbus_message_iter_get_basic(&iter, &rule);
+            dbus_message_iter_get_basic(&array_iter, &rule);
             sudo_req->rules[i] = talloc_strdup(sudo_req->rules, rule);
             if (sudo_req->rules[i] == NULL) {
                 DEBUG(SSSDBG_CRIT_FAILURE, "talloc_strdup failed.\n");
@@ -1593,7 +1594,7 @@ static int be_sudo_handler(struct sbus_request *dbus_req, void *user_data)
                 goto fail;
             }
 
-            dbus_message_iter_next(&iter);
+            dbus_message_iter_next(&array_iter);
         }
 
         sudo_req->rules[rules_num] = NULL;
-- 
1.7.11.7

